utils/testNLC.py

```diff
@@ -0,0 +1,178 @@
+
+        except KeyError:
+            print("class key does not exist!")
+
+def func(args):
+    in_df = None
+    out_df = None
+    test_column = UTTERANCE_COLUMN
+    classifier_id = args.workspace_id
+
+    if args.test_column is not None:  # Test input has multiple columns
+        test_column = args.test_column
+        in_df = pd.read_csv(args.infile, quoting=csv.QUOTE_ALL,
+                            encoding=UTF_8, keep_default_na=False)
+        if test_column not in in_df:  # Look for target test_column
+            raise ValueError(
+                "Test column {} doesn't exist in file.".format(test_column))
+
+        if args.merge_input:  # Merge rest of columns from input to output
+            out_df = in_df
+        else:
+            out_df = in_df[[test_column]].copy()
+            out_df.columns = [test_column]
+
+    else:
+        test_series = pd.read_csv(args.infile, quoting=csv.QUOTE_ALL,
+                                  encoding=UTF_8, header=None, squeeze=True,
+                                  keep_default_na=False)
+        if isinstance(test_series, pd.DataFrame):
+            raise ValueError('Unknown test column')
+        # Test input has only one column and no header
+        out_df = test_series.to_frame()
+        out_df.columns = [test_column]
+
+    # Initial columns for test output
+    for column in test_out_header:
+        out_df[column] = ''
+
+    # Applied coroutines
+    sem = asyncio.Semaphore(args.rate_limit)
+    loop = asyncio.get_event_loop()
+
+    tasks = (fill_df(out_df.loc[row_idx, test_column],
+                     row_idx, out_df, classifier_id,
+                     "apikey", args.iam_apikey, sem, args.url)
+             for row_idx in range(out_df.shape[0]))
+    loop.run_until_complete(asyncio.gather(*tasks))
+
+    loop.close()
+
+    if args.golden_intent_column is not None:
+        golden_intent_column = args.golden_intent_column
+        if golden_intent_column not in in_df.columns:
+            print("No golden intent column '{}' is found in input."
+                  .format(golden_intent_column))
+        else:      
+            out_df[INTENT_JUDGE_COLUMN] = \
+                (in_df[golden_intent_column]
+                    == out_df[PREDICTED_INTENT_COLUMN]).map(BOOL_MAP)
+            out_df[SCORE_COLUMN] = \
+                out_df[INTENT_JUDGE_COLUMN].map({'yes': 1, 'no': 0})
+
+    save_dataframe_as_csv(df=out_df, file=args.outfile)
+
+
+def create_parser():
+    parser = ArgumentParser(
+        description='Test NLC instance using utterance')
+    parser.add_argument('-i', '--infile', type=str, required=True,
+                        help='File that contains test data')
+    parser.add_argument('-o', '--outfile', type=str,
+                        help='Output file path',
+                        default=os.path.join(os.getcwd(), TEST_OUT_FILENAME))
+    parser.add_argument('-w', '--workspace_id', type=str, required=True,
```

Comment:
The "help" can indicate NLC terminology on classifier_id if that is what you are using this for.  Thank you for re-using the same argument switches for ease of compatbility!